Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -61,17 +64,7 @@ class SentrySessionReplay: NSObject {
     }
 
     func start(rootView: UIView, fullSession: Bool) {
-        guard !isRunning else { return }
-        
-        lock.lock()
-        guard !isRunning else {
-            lock.unlock()
-            return
-        }
         displayLink.link(withTarget: self, selector: #selector(newFrame(_:)))
```

Comment:
h: Won't this be a problem? 
You're not invalidating the previous displayLink. 
Which it is really not necessary, we can keep using the same link.
And between `start` and `stop` I believe we should still lock the manipulation of displayLink

Reply:
I'm not sure, since the code never invalidated before my changes. Also, I'm not 100% on this, but I believe a CADisplayLink instance should not be reused after calling `invalidate`. If we wish to "stop"/"start" the same instance, we should be using its `paused` property instead. I'm not sure it's necessary to lock while working directly with the display link property of the SentrySessionReplay instance, because `invalidate`/`paused` are documented as thread safe.

Reply:
It did not invalidate before because if `start` was called when it was already running, the code did nothing and returned early. Now, there is no check anymore, and the code is adding a second callback to displayLink. I'm not sure if displayLink has protection against this.